sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -159,8 +159,9 @@ case class DataSourceAnalysis(conf: CatalystConf) extends Rule[LogicalPlan] {
 
 
     case i @ logical.InsertIntoTable(
-           l @ LogicalRelation(t: HadoopFsRelation, _, _), part, query, overwrite, false)
-        if query.resolved && t.schema.asNullable == query.schema.asNullable =>
```

Comment:
That's why I left a TODO and ask if we do need to rename: https://github.com/apache/spark/pull/15266/files#diff-73bd90660f41c12a87ee9fe8d35d856aL246

INSERT INTO TABLE is resolved by ordinal, so the field names don't matter. How about we don't rename and check `t.schema.map(_.dataType.asNullable) == query.map(_.dataType.asNullable)` here?